code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstanceRemove.java

```diff
@@ -48,7 +49,7 @@ protected void storage(Instance instance, Map<String, Object> data) {
             }
         }
 
-        List<Mount> mounts = getObjectManager().children(instance, Mount.class);
+        List<Mount> mounts = getObjectManager().find(Mount.class, MOUNT.REMOVED, null, MOUNT.INSTANCE_ID, instance.getId());
         for (Mount mount : mounts) {
```

Comment:
@alena1108 per our convo, I verified that deactivating an inactive resource has no ill effect. It just completes with "already done". Also, my thinking with REMOVING was that it would eventually transition to removed and then get excluded from the query and instance.remove would complete.